source/config/profileUpgradeSteps.py

```diff
@@ -493,9 +493,10 @@ def _convertTypingEcho(profile: ConfigObj, key: str) -> None:
 		log.debug(f"'{key}' not present in config, no action taken.")
 		return
 	except ValueError:
-		log.error(f"'{key}' is not a boolean, no action taken.")
+		log.error(f"'{key}' is not a boolean, deleting.")
+		del profile["keyboard"][key]
 		return
 	else:
-		newValue = TypingEcho.EDIT_CONTROLS.value if oldValue else TypingEcho.OFF.value
+		newValue = TypingEcho.ALWAYS.value if oldValue else TypingEcho.OFF.value
```

Comment:
This change from "always" to "only in edit fields" was done on purpose in the original PR.

For character echo, the default (no value in config) is "only in edit fields". Thus, it makes sense that the enabled echo values in old config convert to "only in edit field" too.

Thus I'd revert this change:

```suggestion
		newValue = TypingEcho.EDIT_CONTROLS.value if oldValue else TypingEcho.OFF.value
```